commit 34e77e910cd92c4104cd9d4f250ae03661a7e4fd
Author: bertin <macportsuser@macports.org>
Date:   Thu May 22 23:41:18 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/lamarck/patch-pextlib-passwd-fields.diff

diff --git a/src/pextlib1.0/uid.c b/src/pextlib1.0/uid.c
index ab979978a..aba7e4285 100644
--- a/src/pextlib1.0/uid.c
+++ b/src/pextlib1.0/uid.c
@@ -260,6 +260,13 @@ int getpwuidCmd(ClientData clientData UNUSED, Tcl_Interp *interp, int objc, Tcl_
         Tcl_ListObjAppendElement(interp, reslist, Tcl_NewStringObj(pw->pw_class, -1));
         Tcl_ListObjAppendElement(interp, reslist, Tcl_NewStringObj("expire", -1));
         Tcl_ListObjAppendElement(interp, reslist, Tcl_NewLongObj(pw->pw_expire));
+#else
+        Tcl_ListObjAppendElement(interp, reslist, Tcl_NewStringObj("change", -1));
+        Tcl_ListObjAppendElement(interp, reslist, Tcl_NewLongObj(-1));
+        Tcl_ListObjAppendElement(interp, reslist, Tcl_NewStringObj("class", -1));
+        Tcl_ListObjAppendElement(interp, reslist, Tcl_NewStringObj("", -1));
+        Tcl_ListObjAppendElement(interp, reslist, Tcl_NewStringObj("expire", -1));
+        Tcl_ListObjAppendElement(interp, reslist, Tcl_NewLongObj(-1));
 #endif
         Tcl_SetObjResult(interp, reslist);
         return TCL_OK;
@@ -288,13 +295,25 @@ int getpwuidCmd(ClientData clientData UNUSED, Tcl_Interp *interp, int objc, Tcl_
         return TCL_OK;
 #ifdef __APPLE__
     } else if (strcmp(field, "change") == 0) {
+#ifdef __APPLE__
         Tcl_SetObjResult(interp, Tcl_NewLongObj(pw->pw_change));
+#else
+        Tcl_SetObjResult(interp, Tcl_NewLongObj(-1));
+#endif
         return TCL_OK;
     } else if (strcmp(field, "class") == 0) {
+#ifdef __APPLE__
         Tcl_SetResult(interp, pw->pw_class, TCL_VOLATILE);
+#else
+        Tcl_SetResult(interp, "", TCL_VOLATILE);
+#endif
         return TCL_OK;
     } else if (strcmp(field, "expire") == 0) {
+#ifdef __APPLE__
         Tcl_SetObjResult(interp, Tcl_NewLongObj(pw->pw_expire));
+#else
+        Tcl_SetObjResult(interp, Tcl_NewLongObj(0));
+#endif
         return TCL_OK;
 #endif
     }
